tests/unit/data_formats/datumaro/test_datumaro_format.py

```diff
@@ -236,4 +238,36 @@ def test_inplace_save_writes_only_updated_data_with_transforms(self, test_dir, h
         )
         helper_tc.assertEqual(set(), set(os.listdir(osp.join(test_dir, "images", "c"))))
         helper_tc.assertEqual(set(), set(os.listdir(osp.join(test_dir, "images", "d"))))
+
         compare_datasets(helper_tc, expected, Dataset.import_from(test_dir, format=self.format))
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_export_dm_version(self, test_dir):
+        dataset = Dataset.from_iterable(
+            [DatasetItem(id=1, media=Image(data=np.ones((5, 4, 3))))], media_type=Image
+        )
+
+        legacy_dataset_path = "./tests/assets/datumaro_dataset"
+        for annot_file in glob.glob(
+            osp.join(
+                legacy_dataset_path,
+                DatumaroPath.ANNOTATIONS_DIR,
+                "**",
+                "*" + DatumaroPath.ANNOTATION_EXT,
+            ),
+            recursive=True,
+        ):
+            dm_base = DatumaroBase(path=annot_file)
+            assert (
+                dm_base._get_dm_format_version(dm_base._parsed_anns) == DatumaroBase.LEGACY_VERSION
+            )
+
+        dataset.export(save_dir=test_dir, format="datumaro")
+        for annot_file in glob.glob(
+            osp.join(
+                test_dir, DatumaroPath.ANNOTATIONS_DIR, "**", "*" + DatumaroPath.ANNOTATION_EXT
+            ),
+            recursive=True,
+        ):
+            dm_base = DatumaroBase(path=annot_file)
+            assert dm_base._get_dm_format_version(dm_base._parsed_anns) == DATUMARO_FORMAT_VERSION
```

Comment:
I think this TC is already covered by the existing unit tests so that we can remove this. if you want to make a new TC, I recommend testing the broken version compatibility. For example,

```python
@pytest.fixture
def fxt_wrong_version_dir(fxt_test_datumaro_format_dataset, test_dir):
     dest_dir = osp.join(test_dir, "wrong_version")
     fxt_test_datumaro_format_dataset.export(dest_dir, "datumaro")
     < Somehow edit the annotation file to give some weird version to it >
     yield dest_dir

def test_version_compatibilty(self, fxt_wrong_version_dir):
     with pytest.raises(DatasetImportError):
            Dataset.import_from(fxt_wrong_version_dir, "datumaro")
```

Reply:
This test is checking that
exported dm_dataset files have current datumaro format version or not.

As far as I know, existing unit tests doesn't check this.
if I missed some existing tests, please let me know.

BTW, testing the broken version compatibility is a good idea. I'll add the unit test.

Reply:
If the version compatibility is met, the dataset will be normally parsed without any error, and this is covered by the existing test. I think that code quality is important not only for the source but also for the test code.

Reply:
If somehow datumaro version string is not stored in the exported dataset,

when exporting back, this dataset will be recognized as 'legacy' which is not appreciable but will be parsed well.
Thus I think that we should check whether the exported datumaro has the current datumaro format version well or not.

Reply:
Basically, we need to satisfy the following two things after this PR.
1) Succeed to import the dataset without version info (legacy, what has been used)
2) Succeed to import the dataset with version = 1.0 (to-be)

`2)` will be tested by the existing export and import TCs.

I thought that there has existed a TC for 1) but it doesn't. Sorry for my mistake.
So, it would be better `tests/assets/datumaro_dataset -> tests/assets/datumaro_dataset/legacy`, create a TC to import from `tests/assets/datumaro_dataset/legacy` and `compare_datasets()`. For example,
https://github.com/openvinotoolkit/datumaro/blob/a44fc798760963d4a9510528be284946e3315bac/tests/unit/data_formats/test_common_semantic_segmentation_format.py#L135-L144

It is not appealing to call `dm_base._get_dm_format_version()` because it's a private function.

Reply:
I added `test_version_compatibilty()` and `1)` tests as you suggested. Please have a look